FrameworkInternals/transformDesign.py

```diff
@@ -27,77 +27,146 @@
 from commandMap import getCommand
 import astyleSubstitute
 import subprocess
+import enum
 
-# args:
-# $1    xslt transform filename
-# $2	output filename
-# $3	if 1, will prevent from overwriting output filename, running merge-tool
-# $4	if true, will run astyle on generated file
-# $5	additional params to the transform
+# here we define all XSLT-based transforms of Design known to Quasar
+# first, IDs
+@enum.unique
+class TransformKeys(enum.Enum):
+    AS_SOURCEVARIABLES_H = 1
+    AS_SOURCEVARIABLES_CPP = 2
+    AS_CLASS_H = 3
+    AS_CLASS_CPP = 4
+    AS_INFOMODEL_H = 5
+    AS_INFOMODEL_CPP = 6
+    CONFIGURATION_XSD = 7
+    CONFIGURATOR = 8
+    CONFIG_VALIDATOR = 9
+    DESIGN_VALIDATION = 10
+    UPGRADE_DESIGN = 11
+    CREATE_DIAGRAM_DOT = 12
+    D_ROOT_H = 13
+    D_ROOT_CPP = 14
+    D_BASE_H = 15
+    D_BASE_CPP = 16
+    D_DEVICE_H = 17
+    D_DEVICE_CPP = 18
+    HONKYTONK = 19
+    AS_CMAKE = 20
+    D_CMAKE = 21
+    
+# column IDs
+@enum.unique
+class FieldIds(enum.Enum):
+    KEY = 0
+    XSLT_PATH = 1
+    OUT_PATH = 2 
+    SOURCE_OR_BINARY = 3   
+    CPP_FORMAT = 4
+    OVERWRITE_PROT = 5
+    ADDITIONAL_PARAM = 6
+    
+
+QuasarTransforms = [

```

Comment:
On a general note I very much like this table and the corresponding simplification it allows in the code, it's a nice idea and implementation